src/skmultiflow/meta/accuracy_weighted_ensemble.py

```diff
@@ -0,0 +1,411 @@
+
+
+class AccuracyWeightedEnsemble(StreamModel):
+    """
+    Accuracy Weighted Ensemble or AWE
+
+    Parameters
+    ----------
+    n_estimators: int (default=10)
+        Maximum number of estimators to be kept in the ensemble
+    base_estimator: StreamModel or sklearn.BaseEstimator (default=DecisionTreeClassifier)
+        Each member of the ensemble is an instance of the base estimator
+    window_size: int (default=200)
+        The size of one chunk to be processed
+        (warning: the chunk size is not always the same as the batch size)
+    n_splits: int (default=5)
+        Number of folds to run cross-validation for computing the weight
+        of a classifier in the ensemble
+
+    Notes
+    -----
+    An Accuracy Weighted Ensemble (AWE) [1]_ is an ensemble of classification models in which
+    each model is judiciously weighted based on their expected classification accuracy
+    on the test data under the time-evolving environment. The ensemble guarantees to be
+    efficient and robust against concept-drifting streams.
+
+    References
+    ----------
+    .. [1] Haixun Wang, Wei Fan, Philip S. Yu, and Jiawei Han. 2003.
+    Mining concept-drifting data streams using ensemble classifiers.
+    In Proceedings of the ninth ACM SIGKDD international conference
+    on Knowledge discovery and data mining (KDD '03).
+    ACM, New York, NY, USA, 226-235.
+
+    """
+
+    class WeightedClassifier:
+        """ A wrapper that includes a base estimator and its associated weight
+        (and additional information)
+
+        Parameters
+        ----------
+        estimator: StreamModel or sklearn.BaseEstimator
+            The base estimator to be wrapped up with additional information.
+            This estimator must already been trained on a data chunk.
+        weight: float
+            The weight associated to this estimator
+        seen_labels: array
+            The array containing the unique class labels of the data chunk this estimator
+            is trained on.
+        """
+
+        def __init__(self, estimator, weight, seen_labels):
+            """ Creates a new weighted classifier."""
+            self.estimator = estimator
+            self.weight = weight
+            self.seen_labels = seen_labels
+
+        def __lt__(self, other):
+            """ Compares an object of this class to the other by means of the weight.
+            This method helps to sort the classifier correctly in the sorted list.
+
+            Parameters
+            ----------
+            other: WeightedClassifier
+                The other object to be compared to
+
+            Returns
+            -------
+            boolean
+                true if this object's weight is less than that of the other object
+            """
+            return self.weight < other.weight
+
+    def __init__(self, n_estimators=10, n_kept_estimators=30,
+                 base_estimator=NaiveBayes(), window_size=200, n_splits=5):
```

Comment:
default should be DecisionTreeClassifier according to docstring, right?

Reply:
Oh yeah I'm sorry. In fact, DecisionTree isn't available in skmultiflow, and I don't want to mix up both learners from sklearn and skmultiflow, that's why I switched to NaiveBayes (skmultiflow). I will change the docstring accordingly.

Reply:
This comment has been properly addressed.

Thanks @MinhHuong and @AndreFCruz